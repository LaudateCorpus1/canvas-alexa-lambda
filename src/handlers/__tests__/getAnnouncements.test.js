/*
 * Copyright (C) 2018 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that they will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */
const axios = require("axios");
const MockAdapter = require("axios-mock-adapter");
const createVirtualAlexa = require("../../test_utils/utils.js").createVirtualAlexa;

let mock;
let alexa;

beforeAll(() => {
  alexa = createVirtualAlexa();
  mock = new MockAdapter(axios);
});

afterEach(() => {
  mock.reset();
});

afterAll(() => {
  mock.restore();
});

it("test announcements intent when user has no active courses", async () => {
  mock.onGet("/courses?enrollment_state=active").reply(200, []);
  const result = await alexa.utter("Is there any news");
  expect(result).toMatchSnapshot();
});

it("test announcements intent when user has no announcements", async () => {
  mock
    .onGet("/courses?enrollment_state=active")
    .reply(200, [{ id: 5, name: "temp course name 1" }, { id: 6, name: "temp course name 2" }]);
  mock
    .onGet("/announcements?active_only=true&context_codes[]=course_5&context_codes[]=course_6")
    .reply(200, []);
  const result = await alexa.utter("Is there any news");
  expect(result).toMatchSnapshot();
});

it("test announcements intent when user has announcements", async () => {
  mock
    .onGet("/courses?enrollment_state=active")
    .reply(200, [{ id: 5, name: "temp course name 1" }, { id: 6, name: "temp course name 2" }]);
  mock
    .onGet("/announcements?active_only=true&context_codes[]=course_5&context_codes[]=course_6")
    .reply(200, [
      { id: 5, title: "temp announcement named blah", context_code: "course_6" },
      { id: 6, title: "temp announcement named blee", context_code: "course_5" }
    ]);
  const result = await alexa.utter("Is there any news");
  expect(result).toMatchSnapshot();
});
