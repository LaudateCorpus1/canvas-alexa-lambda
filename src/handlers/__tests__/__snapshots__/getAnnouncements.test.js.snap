// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test announcements intent when user has announcements 1`] = `
SkillResponse {
  "response": Object {
    "outputSpeech": Object {
      "ssml": "<speak>Here are your announcements: In course temp course name 2: temp announcement named blah,
In course temp course name 1: temp announcement named blee. Anything else?</speak>",
      "type": "SSML",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "ssml": "<speak>You can ask to list grades, or check if you have any missing assignments.</speak>",
        "type": "SSML",
      },
    },
    "shouldEndSession": false,
  },
  "sessionAttributes": Object {},
  "userAgent": "ask-node/2.7.0 Node/v8.16.0 ask-sdk-v1adapter/2.7.0",
  "version": "1.0",
}
`;

exports[`test announcements intent when user has no active courses 1`] = `
SkillResponse {
  "response": Object {
    "outputSpeech": Object {
      "ssml": "<speak>You have no active courses. Anything else?</speak>",
      "type": "SSML",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "ssml": "<speak>You can ask to list grades, or check if you have any missing assignments.</speak>",
        "type": "SSML",
      },
    },
    "shouldEndSession": false,
  },
  "sessionAttributes": Object {},
  "userAgent": "ask-node/2.7.0 Node/v8.16.0 ask-sdk-v1adapter/2.7.0",
  "version": "1.0",
}
`;

exports[`test announcements intent when user has no announcements 1`] = `
SkillResponse {
  "response": Object {
    "outputSpeech": Object {
      "ssml": "<speak>You have no announcements. Anything else?</speak>",
      "type": "SSML",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "ssml": "<speak>You can ask to list grades, or check if you have any missing assignments.</speak>",
        "type": "SSML",
      },
    },
    "shouldEndSession": false,
  },
  "sessionAttributes": Object {},
  "userAgent": "ask-node/2.7.0 Node/v8.16.0 ask-sdk-v1adapter/2.7.0",
  "version": "1.0",
}
`;
